### ISSUE TITLE
Reblock 

### ISSUE THREAD

**Issue by mrocklin** [ContinuumIO/dask#58]
We would like to change the block structure of a dask array.  This is particularly important when two dask arrays with differing block structures need to interact.  One way to do this is to make the following function

```python
def reblock(x, blockdims):
    ...

>>> x.blockdims
((10, 10, 10, 10, 10),)

>>> y = reblock(x, ((15, 15, 15, 5),))
>>> y.blockdims
((15, 15, 15, 5),)
>>> (x.compute() == y.compute()).all()
True
```

This looks a bit like a simpler version of slicing.  I think it makes sense to copy @nevermindewe's approach of making a `_slice_1d` function for each dimension of blockdims and then slicing and concatenating as necessary.  

A very simple common case is the promotion of numpy arrays to interact with dask arrays.  A clean solution to this would be to define `Array.__array_wrap__` to call `da.from_array(x, blockshape=x.shape)` and then lean on reblocking to define the block structure after the fact in the `elemwise` call.  This came up in conversation with @shoyer.

**Comment by mrocklin** [comment]
This might be a useful function

```python
old = (10, 10, 10, 10, 10)
new = (25, 5, 20)

>>> f(old, new)
((0, slice(0, 10)), (1, slice(0, 10)), (2, slice(0, 5)), 
 (2, slice(5, 10)), 
 ((3, slice(0, 10)), (4, slice(0, 10)))
```

**Comment by mrocklin** [comment]
Fixed in #113
